pagopa/io-app
# feat: [IOBP-408] New wallet onboarding with custom webview

## ⚠️ Depends on #5189 and https://github.com/pagopa/io-dev-api-server/pull/320 ⚠️

## Short description
This PR changes the new wallet onboarding flow using the custom web view implemented by Authentication team, ensuring an improved and fulfilling user experience.

## List of changes proposed in this pull request
- Removed the previous `WalletOnboardingWebView` component
- Added a new custom hook `useWalletOnboardingWebView` which implements all the business logic to start an onboarding process showing the custom webview. It gives a function `startOnboarding` which accept as input the paymentMethodId and whenever the webview is closed, it emit an `onSuccess` or `onFailure` or `onError` or `onDismiss` event, in this way this hook can be more flexible and reusable in other flows (such as the payment flow)
- Edited the previous `WalletOnboardingStartScreen` to `WalletOnboardingFeedbackScreen` which shows a screen with the relative result screen page

## How to test
- Checkout this PR on dev-server: https://github.com/pagopa/io-dev-api-server/pull/320
- Open the `New wallet playground` section;
- Start the onboarding flow;
- Select a payment method into the list
- It will shows a webview into a modal, select the outcome desidred and check that the feedback page is correct

## Preview

https://github.com/pagopa/io-app/assets/34343582/27852172-576e-464d-be4e-71394f3ec488
